src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -203,8 +205,29 @@ private function finishRequest(Request $request, int $type)
      */
     private function handleThrowable(\Throwable $e, Request $request, int $type): Response
     {
-        $event = new ExceptionEvent($this, $request, $type, $e);
-        $this->dispatcher->dispatch($event, KernelEvents::EXCEPTION);
+        if (!$this->dispatcher instanceof \Symfony\Component\EventDispatcher\EventDispatcherInterface) {
+            @trigger_error('Listening to the "kernel.exception" event is deprecated since Symfony 4.4, listen to "kernel.error" instead.', E_USER_DEPRECATED);
```

Comment:
how can affected people get rid of the deprecation?

Reply:
I think we don't have any other alternative here: ppl that hit this code path must bump to v5.
Good news, this affects ppl that implement `EventDispatcherInterface` from the contracts ONLY. If such an implementation exists (does it), it must be very new, and very rare.